packages/routers/src/__tests__/StackRouter.test.tsx

```diff
@@ -540,10 +540,11 @@ test('ensures unique ID for navigate', () => {
     stale: false,
     type: 'stack',
     key: 'root',
-    index: 1,
+    index: 2,
     preloadedRoutes: [],
     routeNames: ['baz', 'bar', 'qux'],
     routes: [
+      { key: 'bar-test', name: 'bar', params: { foo: 'a' } },
       { key: 'bar', name: 'bar' },
       { key: 'bar-test', name: 'bar', params: { foo: 'a' } },
     ],
```

Comment:
The incoming route will have an id of `bar-a` which does not batch the current route's id: `bar`. Instead of the route at index 0 being moved, a new route should be pushed